### PR TITLE
Ensure that server_keys fetched via a notary server are correctly signed.

### PR DESCRIPTION
**Pull request by richvdh**

### CODE REVIEW

**Comment by richvdh** [comment]
The main change here is in #753b127; the other commit is a follow-up to make the error handling a bit better.

File: synapse/crypto/keyring.py [hunk 1]
```diff
@@ -677,12 +685,6 @@ def get_server_verify_key_v2_direct(self, server_name, key_ids):
             except HttpResponseException as e:
                 raise_from(KeyLookupError("Remote server returned an error"), e)
 
-            if (
-                u"signatures" not in response
-                or server_name not in response[u"signatures"]
-            ):
-                raise KeyLookupError("Key response not signed by remote server")
```
Comment (richvdh) [inline synapse/crypto/keyring.py]: this is now redundant, since we check it properly in `process_v2_response`.

**Comment by erikjohnston** [comment]
I assume that notaries have always signed their responses?

**Comment by richvdh** [comment]
> I assume that notaries have always signed their responses?

They have, but note that it's not the notary's signature that we're checking for here: it is the *origin* server's signature on the json object that is passed on by the notary server.

In theory the notary server should check that the json object is signed by the origin server before it accepts it, but empirically there are a couple of (ancient) key objects in matrix.org's database which aren't signed by the origin server. Also emprically, they don't actually seem to cause a problem with joining either HQ or -dev, so I'm ignoring them.

**Pull request merged**